server/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -626,6 +623,13 @@ public final <K, V> void writeMap(final Map<K, V> map, final Writer<K> keyWriter
         }
     }
 
+    /**
+     * Same as {@link #writeMap(Map, Writer, Writer)} but for {@code String} keys.
+     */
+    public final <V> void writeMap(final Map<String, V> map, final Writer<V> valueWriter) throws IOException {
```

Comment:
naming nit: I'd rather the method name indicated it was for string-keyed maps somehow. `writeStringKeyedMap` is clear but maybe a bit verbose. `writeStringMap` is shortest but maybe implies the values are strings. `writeMapByString` perhaps?

Reply:
Actually I see #96853 already introduced a `readMap()` overload for this case. I'd like to rename them both but that can happen later.